src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -1336,12 +1351,39 @@ codet java_bytecode_convert_methodt::convert_instructions(
 
       exprt var=
         variable(arg0, statement[0], i_it->address, NO_CAST);
+      const irep_idt &var_name=to_symbol_expr(var).get_identifier();
 
       exprt toassign=op[0];
       if('a'==statement[0] && toassign.type()!=var.type())
         toassign=typecast_exprt(toassign, var.type());
 
-      c=code_assignt(var, toassign);
+      code_blockt block;
+      for(size_t i=0; i<stack.size(); i++)
+      {
+        const exprt &stack_entry=stack[i];
+        exprt entry=stack_entry;
+        if(entry.id()==ID_typecast)
```

Comment:
Probably `while`. I certainly try not to introduce double casts, but there are probably still places where they get introduced.